src/Language/PureScript/Docs/Prim.hs

```diff
@@ -225,6 +226,14 @@ fail = primClass "Fail" $ T.unlines
   , "[the Custom Type Errors guide](https://github.com/purescript/documentation/blob/master/guides/Custom-Type-Errors.md)."
   ]
 
+warn :: Declaration
+warn = primClass "Warn" $ T.unlines
+  [ "The Warn type class allows a custom compiler warning to be displayed."
+  , ""
+  , "For more information, see"
+  , "[the Custom Type Errors guide](https://github.com/purescript/documentation/blob/master/guides/Custom-Type-Errors.md)."
```

Comment:
I'm currently writing a quick overview of the feature for this page - I'll submit a PR to the docs repo when this one is merged.